home/tests.py

```diff
@@ -96,3 +94,43 @@ def question_tag_test_data(question_test_data):
     new_pair.tag = test_tag
     new_pair.save()
     return test_tag
+
+
+@pytest.mark.parametrize("question_id, expected_result", [
+    ("1", 1),
+    ("2", 1),
+    ("3", 0),
+    ("4", 0),
+])
+@pytest.mark.django_db
+def test_answers_num_func(question_id, expected_result):
+    question = Question.objects.get(id=question_id)
+    answers_num = question.get_answers_num()
+    assert answers_num == expected_result
+
+
+@pytest.mark.django_db
+@pytest.fixture
+def explore_page_response(client):
+    url = reverse('explore-page')
+    response = client.get(url)
+    return response
+
+
+@pytest.mark.django_db
+def test_explore_page_url(explore_page_response):
+    assert explore_page_response.status_code == 200
+
+
+@pytest.mark.django_db
+def test_explore_page_template(explore_page_response):
+    assert explore_page_response.templates[0].name == "home/explore.html"
+
+
+@pytest.mark.django_db
+def test_explore_page_context(explore_page_response):
```

Comment:
Its better to use client to test views (in this case via the explorer_page_response fixture you've wrote) instead of manually instantiating internal view objects. The way the test is written right now is exposing too many internal implementation details about the view, you won't be able to refactor the view code without changing the test.

Reply:
@ifireball 
I changed it as you requested